sdks/python/apache_beam/io/gcp/datastore/v1/helper.py

```diff
@@ -48,6 +49,7 @@
 # pylint: enable=ungrouped-imports
 
 
+@deprecated(since="v2.4")
```

Comment:
Why are we deprecating these functions?

Reply:
So they encourage `cmp` like behaviour and I didn't see them used internally. This is as a separate commit, I can keep them non-deprecated and just use the rewrite for the internal `cmp` usage.

Reply:
They are in the public API surface, it may have users. Let's not deprecate.

Reply:
For the counterargument see [What's new in Python 3](https://docs.python.org/3/whatsnew/3.0.html#ordering-comparisons): _The cmp() function should be treated as gone, and the __cmp__() special method is no longer supported._

Reply:
I do not follow the counterargument. These two are static functions for comparing datastore keys and paths. Providing these functions should be orthogonal to having cmp functions in the language or not. We can deprecate this if we think that they do not provide value to users, but the decision should not depend on the availability of cmp.

Reply:
We can avoid deprecating them for now and revisit it later.